elements/pfe-content-set/pfe-content-set.ts

```diff
@@ -0,0 +1,341 @@
+// @todo remove eslint disable below
+/* eslint-disable no-console */
+import { LitElement, html as _html, PropertyValues, TemplateResult } from 'lit';
+import { withStatic, unsafeStatic } from 'lit/static-html.js';
+import { customElement, property, state } from 'lit/decorators.js';
+import { classMap } from 'lit/directives/class-map.js';
+// import { ifDefined } from 'lit/directives/if-defined.js';
+
+import { pfelement, observed, bound, initializer } from '@patternfly/pfe-core/decorators.js';
+// import { ComposedEvent } from '@patternfly/pfe-core';
+// import { pfeEvent } from '@patternfly/pfe-core/functions/pfeEvent.js';
+import { Logger } from '@patternfly/pfe-core/controllers/logger.js';
+import style from './pfe-content-set.scss';
+
+import { getRandomId } from '../../core/pfe-core/functions/random.js';
+
+// Helps allow dynamic tags
+const html = withStatic(_html);
+
+// Setup definitions for ContentSet, which is a big part of state in this.contentSets
+interface ContentSet {
+  'index': number,
+  'heading': {
+    'id': string,
+    'element': Element,
+    'text': string|null,
+    'tag': string,
+  }
+  'panel': {
+    'id': string,
+    'element': Element,
+  }
+}
+
+// Setup definition for this.contentSets
+interface ContentSets {
+  [index: string]: ContentSet
+}
+
+@customElement('pfe-content-set') @pfelement()
+export class PfeContentSet extends LitElement {
+  // Setting up default PFE properties
+  static readonly styles = [style];
+  private logger = new Logger(this);
+
+  private _resizeTimeout?: number;
+
+  /** Index of the selected tab */
+  @property({ type: Number, reflect: true, attribute: 'selected-index' }) selectedIndex?: number;
+
+  /** Tab alignment */
+  @property({ type: String, reflect: true, attribute: 'tab-align' }) tabAlign?: 'center';
+
+  /** Tab History */
+  @property({ type: Boolean, reflect: true, attribute: 'tab-history' }) tabHistory = false;
+
+  /** Presentation, e.g. accordion, tabs, vertical tabs */
+  // @todo ask if there is any property validation?
+  @property({ type: String, reflect: true, attribute: 'presentation' }) presentation = 'tabs';
+
+  // Consider revisiting
+  /** Variant */
+  @property({ type: String, reflect: true }) variant: 'wind'|'earth' = 'wind';
+  /** Disclosure */
+  // @todo Consider deprecating, feels like a theme attribute that could have multiple values, not a binary property
```

Comment:
Have to get a little further in the process, but I believe `disclosure` changes the layout of an accordion slightly, so it seems odd to have it be a binary. It could be another option for `variant`, or something else, didn't think on it too hard yet.